lib/web_ui/lib/src/engine/semantics/dialog.dart

```diff
@@ -16,6 +16,39 @@ class Dialog extends PrimaryRoleManager {
     // names its own route an `aria-label` is used instead of `aria-describedby`.
     addFocusManagement();
     addLiveRegion();
+
```

Comment:
should this class be named Route instead? It took me a while to understand a dialog is corresponding to scopesRoute = true

Reply:
The `PrimaryRoleManager` is called `Dialog` because it sets the ARIA `role="dialog"`, which is the unique role identifier. I can rename it, but I'd prefer doing it in a separate PR.

Reply:
Currently "dialog" is the only available ARIA role. A route expressed as a transition to a different screen is expressed on the web by changing the URL (which we already do via `UrlStrategy`) and updating the title of the page (which is expressed via the [Title](https://api.flutter.dev/flutter/widgets/Title-class.html) widget). So I think for routes-that-are-not-dialogs we could improve the situation if the framework communicated the difference somehow. Happy to collaborate on improving this.